docs/notes/02-packaging.md

```diff
@@ -142,6 +117,106 @@ testing notes](05-linters-and-tests.md) for more info.
   
   See the dedicated `pytest` [documentation](https://docs.pytest.org/en/latest/reference/reference.html#ini-options-ref) for detailed informations.
    
-## Resources
-Potentially useful decision tree image from Pyopensci     
-![python-package-tools-decision-tree](https://github.com/flatironinstitute/ccn-template/assets/6643322/2bc2f9a0-c989-4b0a-92d1-ec52693400fc)
+## Manually building and deploying
+
+In this template repo, we use python's `setuptools` to handle the building and deployment of the package. There are many other choices, such as [Poetry](https://python-poetry.org/), but `setuptools` is sufficient for our purposes here, and I find the others confusing. You can look at [PyOpenSci's packaging guide](https://www.pyopensci.org/python-package-guide/package-structure-code/python-package-build-tools.html#) for some more info on the other possibilities (note, however, that I'm not sure how they interact with `cibuildwheel`, if you'll be using that option, which we discuss more below).
+
+Building and deployment are two separate steps. Building creates the source distribution and wheel files described earlier in this note, while deployment puts those files onto PyPI so that users can easily use them to install your package. Typically, they're performed together (there's not really a reason to build if you're not going to deploy).
+
+This section discusses how you would manually build and deploy your package, which is useful for understanding what's going on, but in practice, we handle the building and deployment of our package with Github actions, which take advantage of the fact that Github Action is a PyPI [trusted publisher](https://docs.pypi.org/trusted-publishers/) and thus don't require us to set up API tokens or use username / password for authentication. You can do all the following manually from your local machine, but deploying to PyPI or Test PyPI will require using one of those two other methods for authentication (building does not).
+
+!!! warning
+
+    Note that once you deploy to PyPI, **you cannot undo it.** PyPI can only have one package with a given name and only one upload per version (i.e., you cannot overwrite or re-upload a package with the same version). Therefore, you should be careful about deploying to PyPI and test using Test PyPI first.
+
+### Build
+
+#### Pure python
+
+Assuming your `pyproject.toml` is appropriately set up, you can build your package with the following lines:
+
+```bash
+pip install build # only run the first time
+python -m build --outdir dist/ --sdist --wheel
+```
+
+This will build the source distribution and wheel for your system. If your code is pure python, as discussed above, then you're done! You can double check this by running `ls dist/*.whl`; if your package is pure python, the name will look like `package-version-py3-none-any.whl` (replacing `package` and `version` with the name and version, respectively, of your package). Otherwise, you have native dependencies.
```

Comment:
I'm not sure what it means to say "native dependencies" (from context it seems to mean that it isn't pure Python, but is that all it means? I think it means it depends on OS-specific, compiled code). But maybe just spell this out? E.g., if you have a yaml file that isn't python is it a native dependency (you know what I'm saying).

Reply:
yup, it means "non-python compiled dependencies", which I define on line 16

Reply:
Ah missed that, so it's all good.